src/Microsoft.OData.Client/ALinq/ExpressionWriter.cs

```diff
@@ -526,15 +526,32 @@ internal override Expression VisitConstant(ConstantExpression c)
             else if (m != null && ReflectionUtil.IsSequenceMethod(m.Method, SequenceMethod.Contains))
             {
                 StringBuilder listExpr = new StringBuilder();
-                ODataVersion version = CommonUtil.ConvertToODataVersion(this.uriVersion);
+
                 foreach (object item in (IEnumerable)c.Value)
                 {
                     if (listExpr.Length != 0)
                     {
                         listExpr.Append(UriHelper.COMMA);
                     }
 
-                    string uriLiteral = ODataUriUtils.ConvertToUriLiteral(item, version);
+                    string uriLiteral;
+
+                    try
+                    {
+                        uriLiteral = LiteralFormatter.ForConstants.Format(item);
+                    }
+                    catch (InvalidOperationException)
+                    {
+                        if (this.cantTranslateExpression)
+                        {
+                            // There's already a problem in the parents.
+                            // Return.A up the stack will throw a better exception
+                            return c;
+                        }
+
+                        throw new NotSupportedException(Strings.ALinq_CouldNotConvert(item));

```

Comment:
Will an instance of this expression writer be expected to handle future calls, even in the event of a previous exception? It might make sense to set `cantTranslateExpression` so that the instance remains internally consistent.

Reply:
@corranrogue9 The approach adopted here mirrors the approach taken when escaping special characters in a typical filter expression https://github.com/OData/odata.net/pull/2407/files#diff-7d25f0d0e6eeabce6281ebb5949c72093911c347a158342109816424376c5d73R584